src/labone/nodetree/node.py

```diff
@@ -0,0 +1,863 @@
+
+        if use_cache and path_segments in self._remembered_nodes.keys():
+            return self._remembered_nodes[path_segments]
+
+        result = Node(self, path_segments, subtree_structure, path_aliases)
+        if use_cache:
+            self._remembered_nodes[path_segments] = result
+        return result
+
+    @property
+    def paths(self):
+        return self._paths
+
+
+async def get_nodetree(session: KernelSession, enter_prefix=True):
+    node_tree_session = await NodeTreeManager.create(session)
+    return node_tree_session.get_nodetree(enter_prefix=enter_prefix)
+
+
+class NodeEnumMeta:
+    """Custom Metaclass for NodeEnum.
+
+    Note: Required to enable pickling of a NodeEnum value.
+
+    It simply servers the purpose to recreate a NodeEnum for a given enum
+    value. Since the NodeEnums are created dynamically there is no way recreate
+    a NodeEnum value since python can not find the definition. This class
+    bypasses this problem by providing the functionality to recreate the
+    Enum on the fly.
+
+    Warning: Although the class of the resulting enum object looks and feels
+    the same as the original one it is not. Therefore comparing the `type` will
+    fail. This is however the only limitation.
+    (type(value_old) != type(value_new) but value_old == value_new)
+
+    Args:
+        value: Value of the NodeEnum object that should be created.
+        class_name: Name of the NodeEnum class.
+        names: Mapping of the enum names to their corresponding integer value.
+        module: Should be set to the module this class is being created in.
+    """
+
+    def __new__(  # noqa: D102
+            cls, value: int, class_name: str, names: t.Dict[str, int], module: str
+    ):
+        new_enum = NodeEnum(class_name, names, module=module)
+        return new_enum(value)
+
+
+class NodeEnum(IntEnum):
+    """Custom dynamically picklable IntEnum class.
+
+    The Enum values for a device are created dynamically in toolkit based on
+    the node informations. Since they are not predefined but rather created
+    dynamically, the are not picklable. This custom child class of IntEnum
+    overwrites the reduce function that returns all information required to
+    recreate the Enum class in `NodeEnumMeta`.
+
+    For more information on the reduce functionality and how it is used within
+    the pickle package see
+    [pep307](https://peps.python.org/pep-0307/#extended-reduce-api).
+    """
+
+    # Required for typing
+    def __init__(self, *args, **kwargs):
+        ...
+
+    # Required for typing
+    def __call__(self, *args, **kwargs):  # noqa: D102 # pragma: no cover
+        ...
+
+    def __reduce_ex__(self, _):
+        return NodeEnumMeta, (
+            self._value_,
+            self.__class__.__name__,
+            {key: int(value) for key, value in self.__class__._member_map_.items()},
+            self.__class__.__module__,
+        )
+
+
+class NodeInfo:
```

Comment:
This class only should be acessible for leafe nodes. Therefore we can remove `is_partial` and `contains_wildcards`